program/rust/src/tests/pyth_simulator.rs

```diff
@@ -65,9 +80,92 @@ impl PythSimulator {
             banks_client,
             payer,
             last_blockhash: recent_blockhash,
+            programdata_id: None,
         }
     }
 
+    /// Deploys the oracle program as upgradable
+    pub async fn new_upgradable() -> PythSimulator {
+        let mut bpf_data = read_file(find_file("pyth_oracle.so").unwrap_or_else(|| {
+            panic!("Unable to locate {}", "pyth_oracle.so");
+        }));
```

Comment:
is the build / test process guaranteed to produce this `.so` file before the test runs? That is, is there any possibility of running this test with an outdated binary?

Reply:
Yes `cargo test-bpf` generates this if the source code has changed.